src/components/Examples/ExamplesRenderer.tsx

```diff
@@ -0,0 +1,185 @@
+        classes: {
+          /*
+              For the calcs below:
+              - 350px is the width of the preview (applicable to vertical layouts)
+              - 64px is the padding subtotal within the container at smaller breakpoints
+              - 176px is the padding subtotal within the container at larger breakpoints (vertical layouts)
+              - 160px is the padding subtotal within the container at larger breakpoints (horizontal layouts)
+              */
+          'sp-layout': cn(
+            'w-[calc(100vw-64px)]',
+            isVerticalLayout
+              ? 'md:w-[calc(100vw-64px-350px)] lg:w-[calc(100vw-176px-350px)] xl:w-[calc(1440px-176px-350px)]'
+              : 'lg:w-[calc(100vw-160px)] xl:w-[calc(1440px-160px)]',
+          ),
+          'sp-file-explorer': cn('sm:!w-[172px] sm:min-w-[auto]', isVerticalLayout && 'sm:!h-[400px]'),
+          'sp-editor': cn(isVerticalLayout && 'sm:!h-[400px]'),
+          'sp-preview': cn('!h-[320px] w-full', isVerticalLayout && '!h-[350px] md:!h-full md:!w-[350px]'),
+        },
+        externalResources: [
+          'https://cdn.tailwindcss.com',
+          'https://unpkg.com/franken-ui@2.0.0/dist/css/core.min.css',
+          ...(id === 'pub-sub-history'
+            ? [
+                'https://cdn.jsdelivr.net/npm/uikit@3.16.22/dist/js/uikit.min.js',
+                'https://cdn.jsdelivr.net/npm/uikit@3.16.22/dist/js/uikit-icons.min.js',
+              ]
+            : []),
+        ],
+      }}
+      theme={githubLight}
+      template={determineSandpackTemplate(activeLanguage as LanguageKey)}
+    >
+      <div className="bg-neutral-100 dark:bg-neutral-1200 p-16 rounded-2xl flex flex-col gap-16">
+        <div className="flex gap-4">
+          {Object.keys(rewrittenFiles).map((languageKey) => (
+            <SegmentedControl
+              key={languageKey}
+              variant="subtle"
+              active={activeLanguage === languageKey}
+              rounded
+              leftIcon={`icon-tech-${languageKey}` as IconName}
+              onClick={() => setActiveLanguage(languageKey as LanguageKey)}
+            >
+              {languageKey.charAt(0).toUpperCase() + languageKey.slice(1)}
+            </SegmentedControl>
+          ))}
+        </div>
+        <div className={cn('flex flex-col gap-16 max-w-[calc(100vw-64px)]', isVerticalLayout && 'md:flex-row')}>
+          <CodeEditor
+            theme="light"
+            editor={{
+              className: '',
+              showLineNumbers: true,
+            }}
+          />
+          <div className={cn('flex gap-16 flex-col sm:flex-row', isVerticalLayout && 'md:flex-col')}>
+            <div className="relative flex-1">
+              <SandpackPreview
+                className="rounded-lg overflow-hidden"
+                showOpenInCodeSandbox={false}
+                showRefreshButton
+                {...(id === 'pub-sub-message-encryption' && { startRoute: '?encrypted=true' })}
```

Comment:
[note to collective self] We could always later specify these `startRoute` values in the example data, making things a little more flexible in the process

Reply:
Sounds like a good thing to do if we need a second value for this, rn it's just this encrypted param (and the publisher param, but that's hard set and not example-dependent)